docs/proposals/backup-restore.md

```diff
@@ -1,131 +1,261 @@
+* `--password-stdin`: Read password from stdin.
+* `--identity-token <string>`: Use bearer token for authentication.
+* `--identity-token-stdin`: Read identity token from stdin.
+* `--ca-file <path>`: Path to custom CA certificate file.
+* `--cert-file <path>`: Path to client TLS certificate file.
+* `--key-file <path>`: Path to client private key file.
+* `--resolve <host:port:address[:address_port]>`: Customized DNS for registry.
+* `--debug`: Output debug logs (implies `--no-tty`).
+* `--distribution-spec string`: [Preview] set OCI distribution spec version and API option for target. Options: v1.1-referrers-tag, v1.1-referrers-api
+* `--no-tty`: Disable progress bars.
+
+### User Experience in the CLI
+
+The desired end-to-end user experience of using `oras backup` and `oras restore` to address the identified problems and support the outlined user scenarios is illustrated below.
+
+#### Offline Snapshot for Air-Gapped Environments
+
+Create a snapshot of a sample image `registry-a.k8s.io/kube-apiserver:v1` and its referrer (e.g. signature) for an air-gapped environment:
+
+```console
+oras backup registry-a.k8s.io/kube-apiserver:v1 --include-referrers --output airgap-snapshot.tar
 ```
 
-The backup command will also have the ability to write output to a new compressed tar file where the contents are in a single oci-layout
-format. For example:
+Upon success, the output will be:
+
+```console
+Pulled 1 manifest(s) from registry-a.k8s.io/kube-apiserver:v1
+Found 1 linked referrer(s)
+Included referrers in backup: [application/vnd.cncf.notary.signature] sha256:78833f9c870...
+Exported backup to airgap-snapshot.tar
+Backup completed: 2 manifest(s) written
+```
+
+Transfer the `.tar` file to the air-gapped system via a secured channel. Restore the tarball from local to another registry:
+
+```console
+oras restore registry-b.k8s.io/kube-apiserver:v1 --input airgap-snapshot.tar
+```
+
+Upon success, the output will be:
+
+```console
+Loaded 2 manifest(s) from airgap-snapshot.tar
+Pushed image to registry-b.k8s.io/kube-apiserver:v1
+Pushed linked referrer: [application/vnd.cncf.notary.signature] registry-b.k8s.io/kube-apiserver@sha256:78833f9c870...
+Restore completed successfully
+```
+
+By default, the image and linked referrers are reliably restored to another registry with minimal steps. Users can use the `--exclude-referrers` flag to exclude linked referrers when using `oras restore`.
+
+```console
+$ oras discover registry-b.k8s.io/kube-apiserver:v1
+registry-b.k8s.io/kube-apiserver@sha256:9081a6f83f4febf47369fc46b6f0f7683c7db243df5b43fc9defe51b0471a950
+└── application/vnd.cncf.notary.signature
+    └── sha256:78833f9c870d3b069cdd998cae33b935629399f24743e680ab3bebb90de76589
+        └── [annotations]
+            ├── org.opencontainers.image.created: "2025-06-10T20:25:53Z"
+            └── io.cncf.notary.x509chain.thumbprint#S256: '["xxxxxx"]'
+```
+
+#### Backup Artifacts to a Directory and Restore to Another Registry
```

Comment:
This is a multi-resp scenario. Do you want to keep it or leave it to the next iteration?

Reply:
Multi-repo is complicated as we have to consider how to effectively mapping multiple repos when restoring them to the target registry. Suggest us moving to `v.1.4.0-alpha` and thinking thoroughly about its scenarios and CLI design.